### PR TITLE
Adds garlic

### PR DESCRIPTION
**Pull request by blessedmulligan**
## About The Pull Request
-Adds a new plant and produce, garlic. Garlic seeds are available at the MegaSeed Servitor.
-Garlic contains garlic juice, which (slowly) heals cooks and makes anyone else smell bad.
-Garlic juice stops vampires from draining you, and stuns vampires if it's in their blood.
-Adds new food: garlic bread. It's garlic bread. Crafted with bread slice, garlic, and butter.
-Raw khinkalis now require garlic to craft (khinkalis were the only existing food item with a garlic flavor)

## Why It's Good For The Game

Garlic's fun, opens up a lot of food potential. ~~Also, if we ever get a vampire gamemode, garlic juice can ward off vampires, I guess.~~ if vampires become more important, it helps against them

## Changelog
:cl:
add: Garlic! Heals chefs and makes everyone else stinky. Keeps vampires away.
add: Garlic Bread. Made with a bread slice, butter, and garlic.
Tweak: Khinkalis now require garlic to make, and are slightly more filling.

/:cl:

### CODE REVIEW

**Comment by JJRcop** [comment]
https://github.com/tgstation/tgstation/blob/89769e26bb6391157a1034d737a9aca3138bdf32/code/modules/mob/living/carbon/human/species_types/vampire.dm

**Comment by 123chess456** [comment]
i'm not very good at reading code, but does this ensure that garlic juice causes people to smell on touch and not just consumption? looking to fill a smoke machine with garlic juice to make the whole station smell but if people are wearing internals it doesn't sound like that would work.

**Comment by blessedmulligan** [comment]
> 
> 
> i'm not very good at reading code, but does this ensure that garlic juice causes people to smell on touch and not just consumption? looking to fill a smoke machine with garlic juice to make the whole station smell but if people are wearing internals it doesn't sound like that would work.

I'm not sure if that's necessary. The way garlic works irl is that your body metabolizes sulfur compounds in it to make volatile chemicals that leak out your lungs and skin, making you smell bad. So getting sprayed with garlic juice wouldn't do a whole lot. Definitely not any more than getting sprayed with, say, ground onions.

**Comment by blessedmulligan** [comment]
> 
> 
> https://github.com/tgstation/tgstation/blob/89769e26bb6391157a1034d737a9aca3138bdf32/code/modules/mob/living/carbon/human/species_types/vampire.dm

huh. I assume this is adminfuckery only? never seen it.

**Comment by JJRcop** [comment]
halloween
https://github.com/tgstation/tgstation/blob/89769e26bb6391157a1034d737a9aca3138bdf32/code/modules/mob/living/carbon/human/species_types/vampire.dm#L19-L22

**Comment by blessedmulligan** [comment]
> 
> 
> halloween
> 
so I've got a few months to implement it :^)

**Comment by nemvar** [comment]
Wiznerds can also be vampires

**Comment by ExcessiveUseOfCobblestone** [comment]
No interaction with vampires, cringe.

**Comment by RaveRadbury** [comment]
High potency garlic should cause superpowers/hulk for fat people/chefs.
image

**Comment by blessedmulligan** [comment]
> 
> 
> No interaction with vampires, cringe.
fixed.

**Comment by Tlaltecuhtli** [comment]
eating raw garlic should give you disgust

**Comment by MCterra10** [comment]
garlic bread is the only food I'll ever make as chef now

**Comment by Farquaar** [comment]
Make sure that garlic is thinly sliced
[image](https://i.imgur.com/s3YLwdO.gif?noredirect)

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 1]
```diff
@@ -1,3 +1,9 @@
+/proc/isvampire(mob/living/M)
+	if(!ishuman(M))
+		return FALSE
+	var/mob/living/carbon/human/H = M
+	return(H.dna.species.id == "vampire")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]: https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/is_helpers.dm#L57

File: code/modules/mob/living/carbon/human/species_types/vampire.dm [hunk 2]
```diff
@@ -99,6 +105,10 @@
 				to_chat(victim, "<span class='warning'>[H] tries to bite you, but stops before touching you!</span>")
 				to_chat(H, "<span class='warning'>[victim] is blessed! You stop just in time to avoid catching fire.</span>")
 				return
+			if(victim.reagents && victim.reagents.has_reagent("garlic"))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/vampire.dm]:
```suggestion
			if(victim?.reagents?.has_reagent("garlic"))
```

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -402,6 +402,30 @@
...
+	id = "garlic"
+	description = "Crushed garlic. Chefs love it, but it can make you smell bad."
+	color = "#FEFEFE"
+	taste_description = "garlic"
+	metabolization_rate = 0.05 * REAGENTS_METABOLISM
+
+/datum/reagent/consumable/garlic/on_mob_life(mob/living/carbon/M)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(isvampire(H)) //incapacitating but not lethal. Unfortunately, vampires cannot vomit.
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: you'll be able to replace 414-16 with your helper define.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 3 above)
Comment (blessedmulligan) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: I still need a /carbon/human var to check job and use adjust_hygiene(), right?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: yeah under your helper you'll be able to type as human if true

Comment (blessedmulligan) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: is the current code an ok way of doing that?

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by Grazyn** [comment]
Finally we can make proper pizza marinara

**Comment by OverlyGeneric** [comment]
image

**Pull request merged**